File: 3429-special-array-i/special-array-i.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int n = nums.size();

        for (int i = 1; i < n; i++) {
            if ((nums[i - 1] % 2 == 0 && nums[i] % 2 == 0) or
                (nums[i - 1] % 2 != 0 && nums[i] % 2 != 0)) {
                return false;
            }
        }

        return true;
    }
};

File: 3429-special-array-i/special-array-i.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isArraySpecial = function (nums) {
    let n = nums.length;

    for (let i = 1; i < n; i++) {
        if (nums[i - 1] % 2 === nums[i] % 2) {
            return false;
        }
    }
    return true;
};